src/plugins/foxPage/hooks/useLpApr.ts

```diff
@@ -0,0 +1,51 @@
+import { Fetcher, Token } from '@uniswap/sdk'
+import IUniswapV2Pair from '@uniswap/v2-core/build/IUniswapV2Pair.json'
+import { useEffect, useState } from 'react'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+import {
+  FOX_TOKEN_CONTRACT_ADDRESS,
+  UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+  WETH_TOKEN_CONTRACT_ADDRESS,
+} from '../const'
+import { calculateAPRFromToken0, getEthersProvider } from '../utils'
+import { useContract } from './useContract'
+import { useCurrentBlockNumber } from './useCurrentBlockNumber'
+
+export const useLpApr = () => {
+  const ethersProvider = getEthersProvider()
+  const [lpApr, setLpApr] = useState<string | null>(null)
+  const [loaded, setLoaded] = useState(false)
+  const blockNumber = useCurrentBlockNumber()
+
+  const liquidityContractAddress = UNISWAP_V2_WETH_FOX_POOL_ADDRESS
+  const uniswapLPContract = useContract(
+    ethersProvider,
+    null,
+    liquidityContractAddress,
+    IUniswapV2Pair.abi,
+  )
```

Comment:
Will the useContract hook be used elsewhere? It seems extraneous in this PR when we could just instantiate a new contract here.

`const uniswapLPContract = new Contract(liquidityContractAddress, IUniswapV2Pair.abi, ethersProvider)`

Reply:
Yeah, not for other LPs since this is the only one, but for farming opportunities. I decided to keep this implementation as a hook from the original foxfarm one because of the safety guarantees WRT the possibly undefined provider, but it looks like this can be simplified.

Moving this into draft as I refactor things a bit!

Reply:
Looks like `useContract` is indeed overkill and calling `new Contract` with the right checks in place eveyrwhere we'll need (effectively `useMemo` dependencies and checking for `!uniswapLPContract`) will do the job just as well. Removed it